utility/src/main/java/org/oppia/android/util/profile/NameValidator.kt

```diff
@@ -0,0 +1,32 @@
+package org.oppia.android.util.profile
+
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** Utility to validate profile names */
+@Singleton
+class NameValidator @Inject constructor() {
+  private val letterAndSymbolsRegex by lazy {
+    Regex("^.[${NameValidatorUtil.lettersAndSymbolsRegexString}]+\$")
+  }
+
+  private val noRepeatedAllowedSymbolsRegex by lazy { Regex("[.'-]{2}") }
+
+  /**
+   * Validates names for a profile
```

Comment:
```suggestion
   * Validates a profile name to ensure that it isn't confusingly formatted or contains invalid characters.
```

(May require line wrapping).